http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/utils/NettyClientLogger.java

```diff
@@ -0,0 +1,183 @@
+     *
+     * @param msgSupplier Supplier for the log message
+     */
+    public void debug(Supplier<String> msgSupplier) {
+        debug(msgSupplier, null);
+    }
+
+    /**
+     * Log a DEBUG level message with the given exception.
+     *
+     * @param msgSupplier Supplier for the log message
+     * @param t The throwable to log
+     */
+    public void debug(Supplier<String> msgSupplier, Throwable t) {
+        if (!delegateLogger.isDebugEnabled()) {
+            return;
+        }
+
+        delegateLogger.debug(msgSupplier.get(), t);
+    }
+
+    /**
+     * Log a DEBUG level message including the channel information.
+     *
+     * @param channel The channel for this message is being logged
+     * @param msgSupplier Supplier for the log message
+     */
+    public void debug(Channel channel, Supplier<String> msgSupplier) {
+        debug(channel, msgSupplier, null);
+    }
+
+    /**
+     * Log a DEBUG level message with the given exception and including the channel information.
+     *
+     * @param channel The channel for this message is being logged
+     * @param msgSupplier Supplier for the log message
+     * @param t The throwable to log
+     */
+    public void debug(Channel channel, Supplier<String> msgSupplier, Throwable t) {
+        if (!delegateLogger.isDebugEnabled()) {
+            return;
+        }
+
+        Supplier<String> finalMessage = prependChannelInfo(true, msgSupplier, channel);
+        delegateLogger.debug(finalMessage.get(), t);
+    }
+
+    /**
+     * Log a WARN level message.
+     *
+     * @param msgSupplier Supplier for the log message
+     */
+    public void warn(Supplier<String> msgSupplier) {
+        warn(msgSupplier, null);
+    }
+
+    /**
+     * Log a WARN level message with the given exception.
+     *
+     * @param msgSupplier Supplier for the log message
+     * @param t The throwable to log
+     */
+    public void warn(Supplier<String> msgSupplier, Throwable t) {
+        if (!delegateLogger.isWarnEnabled()) {
+            return;
+        }
+
+        delegateLogger.warn(msgSupplier.get(), t);
+    }
+
+    /**
+     * Log a WARN level message and including the channel information.
+     *
+     * @param channel The channel for this message is being logged
+     * @param msgSupplier Supplier for the log message
+     */
+    public void warn(Channel channel, Supplier<String> msgSupplier) {
+        warn(channel, msgSupplier, null);
+    }
+
+    /**
+     * Log a WARN level message with the given exception and including the channel information.
+     *
+     * @param channel The channel for this message is being logged
+     * @param msgSupplier Supplier for the log message
+     * @param t The throwable to log
+     */
+    public void warn(Channel channel, Supplier<String> msgSupplier, Throwable t) {
+        if (!delegateLogger.isWarnEnabled()) {
+            return;
+        }
+
+        Supplier<String> finalMessage = prependChannelInfo(msgSupplier, channel);
```

Comment:
Even though this is a warning statement, I think we should use the configured log level to determine the verbosity of the channel string. So that even in warnings, if debug level is enabled, we will use the more verbose form.